File: yh_service/src/yh_client.cpp
#include "ros/ros.h"
#include "yh_service/YhSrv.h"
#include<cstdlib>

int main(int argc,char**argv)
{
    ros::init(argc,argv,"yh_client");

    //{"yh_client","a","b"}
    if(argc !=3)
    {
        ROS_INFO("rosrun yh_service yh_client a b");
        ROS_INFO("a, b: int32 number");

        
        return 1;
    }

    ros::NodeHandle nh;
    ros::ServiceClient yh_client=nh.serviceClient<yh_service::YhSrv>("multiply");

    yh_service::YhSrv srv;
    srv.request.a=atoi(argv[1]);
    srv.request.b=atoi(argv[2]);

    if(yh_client.call(srv))
    {
        ROS_INFO("a:%d,b:%d",srv.request.a,srv.request.b);
        ROS_INFO("receive srv: srv.response.result=%d",srv.response.result);


    }
    else{
        ROS_ERROR("Failed to call service");
        return 1;

    }
    return 0;



    return 0;

}

File: yh_service/scripts/yh_client.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-

import rospy
from yh_service.srv import YhSrv,YhSrvRequest
import sys

def multiply_client():
    rospy.init_node("yh_client")

    if len(sys.argv)!=3:
        rospy.loginfo("rosrun yh_sevice yh_client.py a b")
        rospy.loginfo("a,b:int32 number")
        sys.exit(1)
    
    yh_client=rospy.ServiceProxy("multiply",YhSrv)
    
    req=YhSrvRequest()

    req.a=int(sys.argv[1])
    req.b=int(sys.argv[2])

    try:
        res= yh_client(req)
        rospy.loginfo(f"a:{req.a},b{req.b},result:{res.result}")
    except rospy.ServiceException as e:
        rospy.logerr(f"Fail: {e}")
    

if __name__=="__main__":
    multiply_client()



    


    

